### PR TITLE
simplify v2 stake removal

### PR DESCRIPTION
**Pull request by kpeluso**

### CODE REVIEW

File: x/emissions/keeper/msgserver/msg_server_stake.go [hunk 1]
```diff
@@ -57,34 +57,29 @@ func (ms msgServer) AddStake(ctx context.Context, msg *types.MsgAddStake) (*type
...
-			return nil, types.ErrInsufficientStakeToRemove
-		}
 
-		// Push to the stake removal object
-		stakeRemoval.Placements = append(stakeRemoval.Placements, &types.StakePlacement{
-			TopicId: stakePlacement.TopicId,
-			Amount:  stakePlacement.Amount,
-		})
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	stakeToRemove := types.StakeRemoval{
```
Comment (br4e) [inline x/emissions/keeper/msgserver/msg_server_stake.go:72]: wait you can only remove one stake at a time?

File: x/emissions/keeper/msgserver/msg_server_stake.go [hunk 2]
```diff
@@ -118,22 +110,17 @@ func (ms msgServer) ConfirmRemoveStake(ctx context.Context, msg *types.MsgConfir
...
+	coins := sdk.NewCoins(sdk.NewCoin(params.DefaultBondDenom, amountInt))
+	ms.k.SendCoinsFromModuleToAccount(ctx, types.AlloraStakingAccountName, sender, coins)
 
-		// 7. update the stake data structures
-		err = ms.k.RemoveStake(ctx, stakePlacement.TopicId, senderAddr, stakePlacement.Amount)
-		if err != nil {
-			return nil, err
-		}
+	// Update the stake data structures
+	err = ms.k.RemoveStake(ctx, stakeRemoval.Placement.TopicId, sender, stakeRemoval.Placement.Amount)
```
Comment (br4e) [inline x/emissions/keeper/msgserver/msg_server_stake.go:121]: Do we want to set `SetStakeRemoval` to empty after fully removing the stake?

File: x/emissions/keeper/msgserver/msg_server_stake.go (hunk 1 above)
Comment (br4e) [reply x/emissions/keeper/msgserver/msg_server_stake.go:72]: like what if someone wants to just unstake everywhere?

File: x/emissions/keeper/msgserver/msg_server_stake.go (hunk 2 above)
Comment (kpeluso) [reply x/emissions/keeper/msgserver/msg_server_stake.go:121]: think you mean "set the map `stakeRemoval`/`delegateStakeRemoval` to empty after fully removing the stake", to which I'd say "yes"

File: x/emissions/keeper/msgserver/msg_server_stake.go (hunk 1 above)
Comment (kpeluso) [reply x/emissions/keeper/msgserver/msg_server_stake.go:72]: gas is cheap here, should be fine. I don't imagine people to be participating in many topics as each requires specialized knowledge

**Pull request merged**